docs/Task-Deployment.md

```diff
@@ -0,0 +1,45 @@
+## Initial Deployment
+
+This project provides a collection of CloudFormation (CFn) templates to handle the deployment of all of the AWS components necessary to support a RedMine deployment. Additionally, this project includes a "parent" template. The per-service templates can be run individually or their running can handled in a monolithic, coordinated fashion via the parent template. 
+
+It will be necessary for the templates' user to have sufficient privileges to do configure all of the AWS elements present in this template-set. This means the user will need to have privileges to create/modify:
+
+* S3 buckets
+* EC2 security groups
+* EC2 load balancers
+* EFS storage services
+* RDS database clusters and instances
+* EC2 autoscaling groups, launch configurations and instances
+
+If all of these privileges are not present in a single user/role, it will be necessary to execute the templates individually with users/roles appropriate to the individual template.
+
+Whether running via the inividual CFn templates or the "parent" template, the general deployment process/flow looks like:
+
+1. _Execute the S3 bucket setup:_ A private S3 bucket is created with no permissions set.
+1. _Execute the network Security Group setup:_ Three network security groups are set up within the target VPC:
+    1. _Application:_ To be applied to the stack's ASG-managed EC2 instance and to its ALB. Allows port 443 from anywhere and allows port 80 between members of the security group. All other sources/destinations are blocked.
+    1. _Storage:_ To be applied to the stack's ASG-managed EC2 instance and to its EFS share. Allows NFS traffic between the EC2 instance and the EFS mount-targets. All other sources/destinations are blocked.
+    1. _Database:_ To be applied to the stack's ASG-managed EC2 instance and to its RDS service. Allows mysql traffic over standard 3306/TCP between the EC2 instance and its RDS database. All other sources/destinations are blocked.
+1. _Execute the EFS service setup:_ A generic, three-AZ EFS share is configured. The "storage" security group is applied to allow NFS traffic only from the RedMine EC2 host.
+1. _Execute the ELB service setup:_ An application load-balancer is configured. The "application" security group is applied to all HTTPS-based requests from any source and allow HTTP-based communications only between the active RedMine EC2 instance and the load-balancer. A user-selected, ACM-managed SSL certificate is bound to the load-balancer.
+1. _Execute the IAM role setup:_ An IAM policy and attachable instance-role is created. The IAM objects allow unfettered access to the to the S3 bucket from the RedMine-hosting EC2 instances for the purposes of attachment backups. The IAM policy also allows: the AutoScaling service to launch and terminate EC2 instances the the ASG-managed EC2 instaces to use CloudWatch agent for log-export/centralized-logging; and, control of the EC2s via AWS Systems Manager.
```

Comment:
extraneous colon? `allows:`

"the the" -> "the"

Actually, that last sentence just reads kind of weird. Not sure how to edit.